linode/firewalldevice/framework_resource.go

```diff
@@ -18,7 +16,7 @@ func NewResource() resource.Resource {
 		BaseResource: helper.NewBaseResource(
 			helper.BaseResourceConfig{
 				Name:   "linode_firewall_device",
-				IDType: types.StringType,
+				IDType: types.Int64Type,
```

Comment:
Unfortunately I think we need to keep this as a string for the time being due to some weird Crossplane issues (see TPT-2234). I'm not actually sure if that's still the case though 👀

Reply:
As discussed, we can defer the ID type change because it's likely no longer an issue for Crossplane provider, and this is only for changing the type in the base resource which is used for the import state function, it doesn't change the type of the ID in the schema. This is actually a fix because the ID type here mismatched the ID type defined in schema, but since the import state function in the base resource is override in this resource, I think it didn't cause any bug.